### ISSUE TITLE
Twitter Card Type setting not getting populated

### ISSUE THREAD

**Issue by LauraMontgomery** [nystudio107/craft-seomatic#197]
A client has highlighted a small issue to me, were their Twitter Card images weren't been transformed correctly. Some testing on my end and it looks like the Twitter Card Type setting isn't getting pulled from the plugin settings correctly - it is blank:

<img width="587" alt="not-set" src="https://user-images.githubusercontent.com/3043327/44088000-807fc7f4-9fb9-11e8-94fd-a1032078a6a0.png">

Which results in some rather unfortunate, head chopping off:

<img width="365" alt="client-tweet" src="https://user-images.githubusercontent.com/3043327/44088013-92177c1e-9fb9-11e8-89a3-6f0dedead1c4.png">

However, if I go in and manually set this option, in this case to the Summary Card with Large Image option:

<img width="613" alt="manually-set" src="https://user-images.githubusercontent.com/3043327/44088052-ae42ae2c-9fb9-11e8-8141-fbcab80b4437.png">

The image transform works correctly, as you can see from this preview from the Twitter Card Validator:

<img width="455" alt="twitter-card-validator" src="https://user-images.githubusercontent.com/3043327/44088079-c41e7262-9fb9-11e8-8583-030fd90806c6.png">

The option is set both in the Global and Content settings, but just isn't making it through to the entry settings. Like I said, if I set this manually, the transform works as expected, so the client knows to do this for the time-being.

I'm not sure when this may have crept in, as the client is on v3.1.6 but I checked on our agency site, which is running v3.1.13.1, and I'm having the same issue there too.

**Comment by khalwat** [comment]
So update to the latest version, and then go to the Content SEO settings for the affected channels, click on Twitter, and click Save -- it should properly work after that.

**Comment by LauraMontgomery** [comment]
Thanks for getting back to me so quickly. But unfortunately the latest version doesn't appear to have fixed the issue for me. I also tried clearing the caches, then re-applied the setting on the channels, just in case that was adding to the mix.

Interestingly, we still had the dev version of the site available which was running 3.0.24 - so before the critical update - and everything was working as expected then. But after running all the available updates for Craft, SEOmatic, etc. the same issue has developed.

Again, same experience on our agency site.

And I checked with a colleague regarding a site in development that he has been keeping up-to-date, and the issue has appeared there too. Although again, I can't say exactly which version the issue appeared in.

**Comment by khalwat** [comment]
hmmmm. Alrighty, I will have a look.

**Comment by LauraMontgomery** [comment]
Thanks Andrew

**Comment by khalwat** [comment]
Okay so it looks like you're using an SEO Settings field to be able to override the individual settings on a per-entry basis.

The blank entry is there on purpose; if that's selected, it means "no change" -- so it falls back on the Content SEO settings.

If you go to SEOmatic -> Content SEO -> click on the section on question -> Twitter and click on Save, that _should_ take care of the issue for you.

I realize I mentioned this before, but I just want to ensure that we've tried this.

**Comment by khalwat** [comment]
The reason I'm asking is I was able to replicate your issue. Here's an example of an older site (my own in this case) that had deprecated information in the Content SEO settings:

![google chromescreensnapz1323](https://user-images.githubusercontent.com/7570798/44162613-d25e5d80-a08e-11e8-9dbd-2b9fcbbd0530.png)

As you can see, the above image was not cropped as I expected. I went to the appropriate page in Content SEO, and clicked on Save (without having to make any changes) and that fixed it:

![google chromescreensnapz1324](https://user-images.githubusercontent.com/7570798/44162644-e7d38780-a08e-11e8-97ac-a2aff6763547.png)

**Comment by khalwat** [comment]
^ make sure you have updated to SEOmatic 3.1.14 before doing the above.

**Comment by LauraMontgomery** [comment]
Okay, I've managed to find the missing link - I also had to update the Global SEO Twitter settings too. I just re-saved these, then the Content SEO Twitter settings for the affected section, and that has fixed the issue.

Once I re-saved both the Global and Content settings, I seen the Twitter preview change, as per your experience above. And when I checked in an entry, the preview had updated there too, which also was pulled across to the Twitter Card Validator.

Thank you for your help with this, and for clarifying that the blank entry in the drop-down is expected behaviour. Happy to say this is resolved. Cheers again
